src/metabase/api/collection.clj

```diff
@@ -871,22 +873,24 @@
                             {:card-ids (t2/select-pks-set Card :collection_id (u/the-id collection-before-update))}))]
       (api.card/delete-alert-and-notify-archived! alerts))))
 
-#_{:clj-kondo/ignore [:deprecated-var]}
-(api/defendpoint-schema PUT "/:id"
+(api/defendpoint PUT "/:id"
   "Modify an existing Collection, including archiving or unarchiving it, or moving it."
   [id, :as {{:keys [name color description archived parent_id authority_level], :as collection-updates} :body}]
-  {name                                   (s/maybe su/NonBlankString)
-   color                                  (s/maybe collection/hex-color-regex)
-   description                            (s/maybe su/NonBlankString)
-   archived                               (s/maybe s/Bool)
-   parent_id                              (s/maybe su/IntGreaterThanZero)
-   authority_level                        collection/AuthorityLevel}
+  {id              ms/PositiveInt
+   name            [:maybe ms/NonBlankString]
+   color           [:maybe [:re collection/hex-color-regex]]
+   description     [:maybe ms/NonBlankString]
+   archived        [:maybe ms/BooleanValue]
+   parent_id       [:maybe ms/PositiveInt]
+   authority_level collection/AuthorityLevel}
   ;; do we have perms to edit this Collection?
   (let [collection-before-update (api/write-check Collection id)]
     ;; if we're trying to *archive* the Collection, make sure we're allowed to do that
     (check-allowed-to-archive-or-unarchive collection-before-update collection-updates)
+    ;; if authority_level is changing, make sure we're allowed to do that
     (when (and (contains? collection-updates :authority_level)
-               (not= authority_level (:authority_level collection-before-update)))
+               (not= (keyword authority_level) (:authority_level collection-before-update)))
+      (premium-features/assert-has-feature :content-management "Official Collection")
```

Comment:
super nit: I think it feels more natural to say "Official collections is an Enterprise feature...." rather than "Official collection is an Enterprise feature....". Can't explain why though.
```suggestion
      (premium-features/assert-has-feature :content-management "Official Collections")
```

Reply:
Correct me if I'm wrong but I think of "Official Collections" or "Sandboxes" is _a_ feature name, thus I'm using `is` instead of `are`.

Reply:
To my untrained ear "Official collections is a feature" and "Official collections are a feature" are both valid sentences. But "Official collection is a feature" definitely sounds wrong.